NitroxClient/GameLogic/InitialSync/QuickSlotInitialSyncProcessor.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections;
+using System.Collections.Generic;
+using NitroxClient.GameLogic.InitialSync.Base;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Packets;
+
+namespace NitroxClient.GameLogic.InitialSync;
+
+public class QuickSlotInitialSyncProcessor : InitialSyncProcessor
+{
+    public QuickSlotInitialSyncProcessor()
+    {
+        DependentProcessors.Add(typeof(PlayerInitialSyncProcessor));  // the player needs to be configured before we can set quick slots.
+        DependentProcessors.Add(typeof(EquippedItemInitialSyncProcessor)); // we need to have the items spawned into our inventory before we can quick slot them.
+    }
+
+    public override IEnumerator Process(InitialPlayerSync packet, WaitScreen.ManualWaitItem waitScreenItem)
+    {
+        int nonEmptySlots = 0;
+
+        Dictionary<NitroxId, InventoryItem> inventoryItemsById = GetItemsById();
+ 
+        for (int i = 0; i < packet.QuickSlotsBindingIds.Length; i++)
+        {
+            waitScreenItem.SetProgress(i, packet.QuickSlotsBindingIds.Length);
+
+            NitroxId id = packet.QuickSlotsBindingIds[i];
+
+            if (id != null && inventoryItemsById.TryGetValue(id, out InventoryItem inventoryItem) )
```

Comment:
If the id is null, we should probably still unbind the slot because by default, there'll be a flare in there

Reply:
All slots are unbound by default when starting a new game.  The issue above is that we only patch QuickSlot.Bind and not QuickSlot.Unbind (thus the server thinks we still have everything full).  I can add this new patch.

Reply:
I take it back, unbinds were properly registered with the server.  it was just the initial equipment sync items.  added the unbind line here on null and tools seems to work.